models/CoronSegmentator/scripts/cardiac_segmentation/cardiac_seg.py

```diff
@@ -0,0 +1,188 @@
+            logger.error(f"Error processing {name}: {str(e)}")
+            raise
+        finally:
+            gc.collect()
+
+    def _process_segmentation_results(self, seg_nrrd_path, output_dir):
+        """Process all segments in parallel and generate STL files."""
+        try:
+            seg_image = sitk.ReadImage(seg_nrrd_path)
+
+            seg_array = np.unique(sitk.GetArrayFromImage(seg_image))
+            unique_labels = set(seg_array)
+            segments_to_process = [s for s in self.segments if s["label"] in unique_labels]
+            logger.info(
+                f"Processing {len(segments_to_process)} segments out of {len(self.segments)} defined segments for {os.path.basename(seg_nrrd_path)}"
+            )
+
+            with concurrent.futures.ThreadPoolExecutor(
+                max_workers=min(int(os.cpu_count() * 0.75), len(segments_to_process))
+            ) as executor:
+                futures = [
+                    executor.submit(self._process_single_segment, seg_image, segment, output_dir)
+                    for segment in self.segments
+                ]
+
+                for future in concurrent.futures.as_completed(futures):
+                    try:
+                        future.result(timeout=300)
+                    except TimeoutError as te:
+                        future.cancel()
+                    except Exception as e:
+                        logger.error(f"processing task for {futures[future]} failed: {str(e)}")
+            # for segment in self.segments:
+            #    self._process_single_segment(seg_image, segment, output_dir)
+
+            FileConversion.convert_nrrd_to_nii(
+                seg_nrrd_path, os.path.join(self.output_path, f"{Path(seg_nrrd_path).stem}.nii")
+            )
+
+            os.remove(seg_nrrd_path)
+            del seg_image, seg_array
+            gc.collect()
+        except Exception as e:
+            raise RuntimeError(f"Result processing failed: {str(e)}")
+
+    def run(self):
+        """Execute the complete processing pipeline."""
+        start_time = time.time()
+        logger.info(f"Starting cardiac segmentation for {os.path.basename(self.input_path)}")
+
+        self.create_directory(self.output_path)
+
+        with (
+            tempfile.TemporaryDirectory(prefix=f"CardiacSeg_Input_{self.instance_id}_") as input_tmp_dir,
+            tempfile.TemporaryDirectory(prefix=f"CardiacSeg_Output_{self.instance_id}_") as output_tmp_dir,
+        ):
+            # try:
+            # Step 1: Convert input to NRRD
+            input_nrrd = self._convert_to_nrrd(self.input_path, input_tmp_dir)
+            # Step 2: Run segmentation
+            seg_nrrd = self._run_segmentation(input_nrrd, output_tmp_dir)
+            # Step 3: Process results
+            self._process_segmentation_results(seg_nrrd, output_tmp_dir)
+            # except Exception as e:
+            #     logger.error(f"Processing failed: {str(e)}")
+            #     raise
```

Comment:
Could you please remove the comment code here or perhaps include the try-catch logic is also make sense.